src/traffic-editor.md

```diff
@@ -1,3 +1,522 @@
+```
+
+### Adding a traffic lane
+One of the most important tools in the `traffic_editor` GUI is the _Add lane_
+tool available in the _Traffic_ edit mode. The allowable motions of each fleet
+operating in the facility is conveyed through its respective Graph which
+consists of waypoints and connecting lanes. The `traffic` tab in the `Sidebar`
+has a default of nine Graphs for nine different fleets. To annotate lanes for a
+graph, say Graph 0, select the Graph from the `traffic` tab and click the _Add
+lane_ tool. Lanes for this graph can be drawn by clicking vertices to be
+connected. If a vertex is not present, it will automatically be added.
+Properties may be assigned to each vertex as described in the preceding section.
+To issue tasks to waypoints that require the robot to terminate at any waypoint,
+a name must be assigned to the waypoint. 
+
+![](images/traffic_editor/add_lane.png)
+
+Each Graph has a unique color for its lanes and their visibility may be
+toggled using the checkbox in the `traffic` tab. A lane that is defined between
+two waypoints may be configured with these additional properties
+* **bidirectional:** if `true`, the `rmf_fleet_adapter` will plan routes for its
+  robot assuming the lanes can be traversed in both directions. Lanes that are
+  not bidirectional have arrows indicating their directionality (indigo lanes in
+  figure above).
+* **graph_idx**: the Graph number a lane corresponds to.
+* **orientation**: constrain the lane to make the robot travel in `forward` or `backward` orientation.
+
+While lifts that move between levels are now supported in the `traffic_editor`,
+the **demo_mock_floor_name** and **demo_mock_lift_name** properties were
+originally engineered to showcase shared lift access in a single floor
+demonstration environment with a "mock" lift that receives lift commands and
+transmits lift states but does not actually move between any different floors in
+a building. However, as there may be interest such functionality for testing
+single-floor hardware setups that seek to emulate multi-floor scenarios, these
+properties were retained.
+
+* **demo_mock_floor_name**: name of the floor that the robot is on while
+  traversing the lane
+* **demo_mock_lift_name**: name of the lift that is being entered or exited
+  while the robot traverses the lane
+
+To further explain these properties, consider this representation of a
+navigation graph where numbers are waypoints and letters are lanes:
+```
+1 <---a---> 2 <---b---> 3
+
+Waypoint 1 is on floor L1
+Waypoint 2 is inside the "lift" named LIFT001
+Waypoint 3 is on floor L3
+The properties of edge "a" are:
+    bidirectional: true
+    demo_mock_floor_name: L1
+    demo_mock_lift_name: LIFT001
+The properties of edge "b" are:
+    bidirectional: true
+    demo_mock_floor_name: L3
+    demo_mock_lift_name: LIFT001
+```
+
+If the robot is to travel from 1 to 3, the `rmf_fleet_adapter` will request for
```

Comment:
```suggestion
If the robot is to travel from level 1 to 3, the `rmf_fleet_adapter` will request for
```

Reply:
Modified to "from waypoint 1 to waypoint 3" for better clarity. https://github.com/osrf/ros2multirobotbook/pull/20/commits/7246171e3476fc0bbbb5402dae5dd21e8d163165